tools/azure-pipelines/jobs-template.yml

```diff
@@ -121,15 +128,34 @@ jobs:
     continueOnError: true # continue the build even if the cache fails.
     condition: not(eq('${{parameters.test_pool_definition.name}}', 'Default'))
     displayName: Cache Maven local repo
+
   - script: |
       echo "##vso[task.setvariable variable=JAVA_HOME]$JAVA_HOME_11_X64"
       echo "##vso[task.setvariable variable=PATH]$JAVA_HOME_11_X64/bin:$PATH"
     displayName: "Set to jdk11"
     condition: eq('${{parameters.jdk}}', 'jdk11')  
+
   - script: sudo sysctl -w kernel.core_pattern=core.%p
     displayName: Set coredump pattern
+
   # Test
-  - script: STAGE=test ${{parameters.environment}} ./tools/azure-pipelines/azure_controller.sh $(module)
+  - script: |
+      ./tools/azure-pipelines/unpack_build_artifact.sh
+      export DEBUG_FILES="$AGENT_TEMPDIRECTORY/debug_files"
```

Comment:
why isn't all this put into a script?

Reply:
I considered it, but I had the feeling that these steps are still short enough to keep them in the template.yml.
If you think it would improve readability,  can move them into script.

Reply:
I figured as much.

I'd say that if it is not just calling script A,B,C, then add a separate script for it. What we absolutely want to prevent is the scripts being spread out over several places.
TBH there's probably even merit in having each step call at most 1 script.

Reply:
Thanks for your feedback. I will drastically shorten the `- script: |` sections.